### PR TITLE
Security check before using local storage

### PR DESCRIPTION
**Pull request by pcoenen**

### CODE REVIEW

File: views/home.php [hunk 1]
```diff
@@ -204,14 +205,16 @@ function showSelectedCourses(){
...
         $(document).ready(function(){
-            var courses = localStorage.getItem("courses");
-            if (courses){
-                var courses = courses.split(",");
-                courses.forEach(function(course, index){
-                    toggleCourseCSV(course);
-                    $("#"+course).addClass("active");
-                });
-            }
+        	if (typeof(Storage) !== "undefined") {
```
Comment (MidasLamb) [inline views/home.php]: why arent you checking typeof(localStorage)?

**Review by MidasLamb — COMMENTED** [review]
Why not typeof(localStorage)?

File: views/home.php (hunk 1 above)
Comment (pcoenen) [reply views/home.php]: W3 reference uses this notation?

**Comment by pcoenen** [comment]
Changed it to local storage

File: views/home.php [hunk 2]
```diff
@@ -24,6 +24,7 @@
         <div class="row">
             <div class="col-md-4 order-md-2">
                 <h2>Export to ICS</h2>
+                <p style="color:red" id="warning"></p>
```
Comment (MidasLamb) [inline views/home.php]: @StijnCaerts Maybe it's better to use a [Bootstrap alerts](https://getbootstrap.com/docs/4.0/components/alerts/)? Style is your expertise ;)

File: views/home.php [hunk 3]
```diff
@@ -112,6 +113,8 @@ function toggleCourseCSV(course){
             var courseCSV = newLink.split("?")[1].split("=")[1];
             if (typeof(Storage) !== "undefined") {
 	    		localStorage.setItem("courses", courseCSV);
+	    	} else {
+	    		document.getElementById("warning").innerHTML = = "Because your browser doesn't support local storage, the selected subjects will not be saved."
```
Comment (MidasLamb) [inline views/home.php]: Why not always have the text and toggle the display property?

**Review by MidasLamb — COMMENTED** [review]
Some small remarks

File: views/home.php (hunk 3 above)
Comment (MidasLamb) [reply views/home.php]: and i doubt the ```= =``` is correct?

**Comment by pcoenen** [comment]
That's also an option, but thé result is the same, so it doesn't matter in my opinion. 

Will change the = = sorry my mistake

**Comment by MidasLamb** [comment]
@pcoenen, the visual result is the same, but internally, updating the dom is way more demanding, look at the difference in this test: 
image

**Comment by pcoenen** [comment]
Changed it

**Review by MidasLamb — COMMENTED** [review]
Isn't the warning always visible? You're changing the style from the standard ```inline-block```(i think) to ```block```,  but i can't seem to see where the warning is hidden if you do have ```localStorage```

**Comment by pcoenen** [comment]
I noticed it, was already fixing it

**Comment by pcoenen** [comment]
fixed

**Comment by MidasLamb** [comment]
@pcoenen, can i suggest something like this:
image
image

**Pull request closed without merging**